cloudinit/subp.py

```diff
@@ -144,6 +144,23 @@ def _indent_text(
         return text.rstrip(b"\n").replace(b"\n", b"\n" + b" " * indent_level)
 
 
+def is_invalid_command(args: Union[List[str], List[bytes]]):
+    """check argument types to ensure that subp() can run the argument
+
+    Throw a user-friendly exception which explains the issue.
+
+    args: list of arguments passed to subp()
+    raises: ProcessExecutionError with information explaining the issue
+    """
+    for component in args:
+        # if already bytes, or implements encode(), then it should be safe
+        if not (isinstance(component, bytes) or hasattr(component, "encode")):
+            LOG.warning("Running invalid command %s", args)
```

Comment:
Nit: `:` after command or putting the command in quotes might be clearer.

Reply:
+1, I'll go with `:` since I would expect this usually to be an array-based invalid argument, so extra quotes would be messy.